gluten-core/src/main/scala/org/apache/gluten/initializer/CodedInputStreamClassInitializer.scala

```diff
@@ -34,8 +38,14 @@ object CodedInputStreamClassInitializer {
       val field: Field = clazz.getDeclaredField("defaultRecursionLimit")
       field.setAccessible(true)
       // Enlarge defaultRecursionLimit whose original value is 100.
-      field.setInt(null, 100000)
+      field.setInt(null, newDefaultRecursionLimit)
+      logInfo(
+        s"The defaultRecursionLimit in protobuf has been increased to $newDefaultRecursionLimit")
     } catch {
+      // The class is shaded during package phase (see package/pom.xml). We tolerate
+      // this ClassNotFoundException in mvn test.
+      case _: ClassNotFoundException =>
+        logWarning("The defaultRecursionLimit in protobuf has not been modified.")
```

Comment:
Curious why don't we support both cases? Say we either find `com.google.protobuf.CodedInputStream`, or find `org.apache.gluten.shaded.com.google.protobuf.CodedInputStream`, or raise an error.

Reply:
@zhztheplayer, just updated the pr to allow loading the original class. Thanks!